lib/puppetfactory.rb

```diff
@@ -85,9 +85,22 @@ def settings.puppetfactory
 
   # set the currently active user. This should probably be a PUT action.
   get '/users/active/:username' do |username|
+    username ||= myParams[user]
+
+    @users ||= load_users()
+
+    unless @users.include?(username)
+      return {"status" => "failure"}.to_json
+    end
+
     session[:username] = username
     {"status" => "ok"}.to_json
-    redirect '/'
+    redirect '/'    
+  end
+  
+  get '/users/deactive/:username' do |username|
```

Comment:
This doesn't really need to take a username parameter, since it's just deleting the cookie regardless of which username is actually recorded there.

Reply:
We don't want to delete the whole cookie, since I think it is the same cookie that is storing the admin session.